mkosi/backend.py

```diff
@@ -76,6 +76,10 @@ def roundup(x: int, step: int) -> int:
 class MkosiException(Exception):
     """Leads to sys.exit"""
 
+    def __init__(self, *args: Any, returncode: int = 1, **kwargs: Any) -> None:
```

Comment:
```suggestion
    def __init__(self, *, returncode: int = 1) -> None:
```
I think keeping variable and keyword arguments apart is something we should enforce in our signatures (unfortunately we our minimum Python requirement doesn't yet support `/`). This would leave open adding any variable (or keyword) args later on andI think it's generally more readable (the constructor below can be called without arguments).

Reply:
Yeah, but I thought it'd still be useful to use MkosiException as a general purpose exception as well. If we don't pass any arguments to the constructor we lose that.

Reply:
Ah, right, then I'd go for 
```suggestion
    def __init__(self, *args: Any, *, returncode: int = 1, **kwargs: Any) -> None:
```
while meaning the not yet available
```suggestion
    def __init__(self, *args: Any, /, *, returncode: int = 1, **kwargs: Any) -> None:
```

Reply:
I get a pyright error telling me the keyword only argument separator isn't allowed after `*args` when trying to apply this suggestion.